src/background/messaging/rpc-message-handler.ts

```diff
@@ -36,6 +40,59 @@ export async function rpcMessageHandler(message: WalletRequests, port: chrome.ru
       break;
     }
 
+    case 'acceptOffer': {
+      if (!message.params) {
+        chrome.tabs.sendMessage(
+          getTabIdFromPort(port),
+          makeRpcErrorResponse('acceptOffer', {
+            id: message.id,
+            error: {
+              code: RpcErrorCode.INVALID_PARAMS,
+              message:
+                'Invalid offer parameters',
+            },
+          })
+        );
+        break;
+      }
+
+        if (message.params.bitcoinContractOffer.includes('Invalid state: Not enough fund in utxos')) {
+          chrome.tabs.sendMessage(
+            getTabIdFromPort(port),
+            makeRpcErrorResponse('acceptOffer', {
+              id: message.id,
+              error: {
+                code: RpcErrorCode.INVALID_REQUEST,
+                message:
+                  'The counterparty does not have enough funds to complete the offer',
+              },
+            })
+          );
+          break;
+        }
+
+      const { urlParams, tabId } = makeSearchParamsWithDefaults(port, [
+        ['bitcoinContractOffer', message.params?.bitcoinContractOffer!],
+        ['counterpartyWalletURL', message.params?.counterpartyWalletURL!],
+        ['counterpartyWalletName', message.params?.counterpartyWalletName!],
+        ['counterpartyWalletIcon', message.params?.counterpartyWalletIcon!],
+        ['requestID', message.id],
```

Comment:
stylized in the other rpc events as 

```suggestion
        ['requestId', message.id],
```